VERSIONING.md

```diff
@@ -0,0 +1,145 @@
+changes. For example,
+[this](https://github.com/open-telemetry/opentelemetry-dotnet/tree/master/src/OpenTelemetry.Instrumentation.AspNetCore/.publicApi)
+shows the public per target framework for the
+`OpenTelemetry.Instrumentation.AspNetCore` package.
+
+Since no stable version has been released so far, every API is listed in the
+"Unshipped.txt" file. Once 1.0.0 is shipped, it'll be moved to "Shipped.txt"
+file.
+
+## Packaging
+
+OpenTelemetry is structured around signals like Traces, Metrics, Logs, Baggage
+etc. OpenTelemetry .NET mostly provides multiple signals in the same package.
+i.e the `OpenTelemetry` package will consist of SDK components from *all* the
+signals. Instrumentations also follow the same model - for example, there will
+be a single package `OpenTelemetry.Instrumentation.AspNetCore` for ASP.NET Core
+instrumentation, which produces Traces, Metrics and handles propagation, instead
+of separate packages.
+
+Due to the fact that OpenTelemetry .NET relies on .NET runtime to provide many
+instrumentation API, API package is naturally split into multiple. i.e
+`System.Diagnostics.DiagnosticSource` provides the Tracing API and
+`Microsoft.Extensions.Logging.Abstractions` provide the Logging API. In the
+future, APIs for Metrics, Baggage and Propagators will also come from .NET
+runtime itself.
+
+## Experimental Signals
+
+Due to the fact that OpenTelemetry specification for different signals achieve stability at different pace, we'll need to deal with shipping experimental, non-stable signals. As of Dec 2020, Metrics and Logs specs have not been declared stable.
+
+### Approach 1 - Using Obsolete attribute
+
+Note: This is the currently implemented approach.
+
+Any experimental signal which is shipped as part of a normal package will be
+marked
+[Obsolete](https://docs.microsoft.com/dotnet/api/system.obsoleteattribute).
+(Ideally we need an "Experimental" attribute, but .NET has only built-in support
+for "Obsolete", so we are leveraging it. Using "Obsolete" code results in
+compile time warnings to caution users.) This allows an experimental signal to
+co-exist in the same package as other, non-experimental signals. For example,
+the OpenTelemetry 1.0.0 package will consist of Traces and Metrics. As Metrics
+signal is not ready for stable release, every method dealing with Metrics will
+be marked "Obsolete". Once Metrics signal gets stable, it'll be added to a
+release with minor version bump, say 1.2.0. The "Obsolete" Metric methods can be
+removed later with a major version bump.
+
+### Example with Approach1
+
+Following shows an example on how the `OpenTelemetry` package versioning works
+with this approach:
+
+`OpenTelemetry` 0.7.0-beta1 release : Pre-release, no API guarantees.
+
+`OpenTelemetry` 1.0.0-RC1 release : Pre-release, no API guarantees, but more
+stable than beta.
+
+`OpenTelemetry` 1.0.0 release : Stable release consisting of Traces, Propagators
```

Comment:
There is a variation of Approach1 - ship 1.0.0 release by removing metrics entirely.